soccer/planning/PlanRequest.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <planning/MotionCommand.hpp>
+#include <planning/MotionConstraints.hpp>
+#include <planning/MotionInstant.hpp>
+#include <planning/Path.hpp>
+#include "SystemState.hpp"
+#include "planning/DynamicObstacle.hpp"
+
+#include <map>
+#include <memory>
+#include "RobotConstraints.hpp"
+
+namespace Planning {
+
+/// The PlanRequest encapsulates all information that the planner needs to know
+/// about an individual robot in order to generate a path for it.
+struct PlanRequest {
```

Comment:
Since this is the new PlanRequest replacement, could you write a bit of documentation for the fields in the struct?

The format with doxygen should be (documented [here](http://www.stack.nl/~dimitri/doxygen/manual/docblocks.html)):

```
 int8_t priority     /**< Priority of this plan request. Lower is higher (?) */
```